codeowners/src/lib.rs

```diff
@@ -1,455 +1,28 @@
-//! Codeowners provides interfaces for resolving owners of paths within code
-//! repositories using
-//! Github [CODEOWNERS](https://help.github.com/articles/about-codeowners/)
-//! files
-//!
-//! # Examples
-//!
-//! Typical use involves resolving a CODEOWNERS file, parsing it,
-//! then querying target paths
-//!
-//! ```no_run
-//! extern crate codeowners;
-//! use std::env;
-//!
-//! fn main() {
-//!   if let (Some(owners_file), Some(path)) =
-//!      (env::args().nth(1), env::args().nth(2)) {
-//!      let owners = codeowners::from_path(owners_file).unwrap();
-//!      match owners.of(&path) {
-//!        None => println!("{} is up for adoption", path),
-//!        Some(owners) => {
-//!           for owner in owners {
-//!             println!("{}", owner);
-//!           }
-//!        }
-//!      }
-//!   }
-//! }
-//! ```
-#![allow(missing_docs)]
+mod github;
 
-#[cfg(test)]
-#[macro_use]
-extern crate pretty_assertions;
+use std::{io::Read, path::Path};
 
-use glob::Pattern;
-use lazy_static::lazy_static;
-use regex::Regex;
-use std::{
-    fmt,
-    fs::File,
-    io::{BufRead, BufReader, Read},
-    path::{Path, PathBuf},
-    str::FromStr,
-};
+pub use github::{GitHubOwner, GitHubOwners};
 
-const CODEOWNERS: &str = "CODEOWNERS";
+pub trait OwnersOfPath {
+    type Owner;
```

Comment:
split primary public interface of this crate into traits to allow implementing a similar interface for each CODEOWNERS provider